From f7e5515272ec62735a283addc642a3930ee5e943 Mon Sep 17 00:00:00 2001
From: Antonio Alvarez Feijoo <antonio.feijoo@suse.com>
Date: Mon, 3 Mar 2025 15:22:14 +0100
Subject: [PATCH] chore(multipath): remove `rd_NO_MULTIPATH` kernel command
 line option

Deprecated since 778b3543609d8c9d32df7111229f4072d00d02f0 (Nov 25, 2014).

(cherry picked from commit 954420374a63be32f6ece5fc1476ba18163da1f7)

Related: RHEL-96106
---
 man/dracut.cmdline.7.asc                 | 2 --
 modules.d/90multipath/multipathd.service | 1 -
 modules.d/90multipath/multipathd.sh      | 2 +-
 3 files changed, 1 insertion(+), 4 deletions(-)

diff --git a/man/dracut.cmdline.7.asc b/man/dracut.cmdline.7.asc
index abe1ec3f..697f81d9 100644
--- a/man/dracut.cmdline.7.asc
+++ b/man/dracut.cmdline.7.asc
@@ -1383,8 +1383,6 @@ rd_NO_MD:: rd.md=0
 
 rd_MD_UUID:: rd.md.uuid
 
-rd_NO_MULTIPATH: rd.multipath=0
-
 rd_NFS_DOMAIN:: rd.nfs.domain
 
 iscsi_initiator:: rd.iscsi.initiator
diff --git a/modules.d/90multipath/multipathd.service b/modules.d/90multipath/multipathd.service
index e8182533..1ff7201a 100644
--- a/modules.d/90multipath/multipathd.service
+++ b/modules.d/90multipath/multipathd.service
@@ -11,7 +11,6 @@ Conflicts=shutdown.target
 Conflicts=initrd-cleanup.service
 ConditionKernelCommandLine=!nompath
 ConditionKernelCommandLine=!rd.multipath=0
-ConditionKernelCommandLine=!rd_NO_MULTIPATH
 ConditionKernelCommandLine=!multipath=off
 ConditionVirtualization=!container
 
diff --git a/modules.d/90multipath/multipathd.sh b/modules.d/90multipath/multipathd.sh
index 385a95fb..62649003 100755
--- a/modules.d/90multipath/multipathd.sh
+++ b/modules.d/90multipath/multipathd.sh
@@ -5,7 +5,7 @@ if [ "$(getarg rd.multipath)" = "default" ] && [ ! -e /etc/multipath.conf ]; the
     mpathconf --enable
 fi
 
-if getargbool 1 rd.multipath -d -n rd_NO_MULTIPATH && [ -e /etc/multipath.conf ]; then
+if getargbool 1 rd.multipath && [ -e /etc/multipath.conf ]; then
     modprobe dm-multipath
     multipathd -B || multipathd
     need_shutdown

